tests/session.test.ts

```diff
@@ -546,6 +546,86 @@ test.describe('Session', () => {
 		);
 	});
 
+	test('handles abort and navigation during completion', async ({ page }) => {
+		const aiMessage = page.locator('article', { hasText: 'Assistant' });
+		const errorMessage = page.getByText(
+			"Can't stop completion until the first response from Ollama is received"
+		);
+
+		async function setupAndRunPrompt() {
+			// Setup
+			await page.goto('/');
+			await mockTagsResponse(page);
+			await chooseModelFromSettings(page, MOCK_API_TAGS_RESPONSE.models[0].name);
+			await page.getByText('Sessions', { exact: true }).click();
+			await page.getByTestId('new-session').click();
+
+			// Submit prompt
+			await promptTextarea.fill('Test prompt');
+			await page.getByText('Run').click();
+		}
+
+		// Mock a slow response
+		await page.route('**/api/chat', async (route) => {
+			await new Promise((resolve) => setTimeout(resolve, 1000)); // Simulate pending state
+			await route.fulfill({ status: 200, body: '{"message": {"content": "Loading..."}}' });
+		});
```

Comment:
> As far as I understand, it should work, but it doesn't. It fails because, at line 609, it expects the completion to be started, but it is not. I think it has to be something about the stream response emulated at line 591.

When I run the test I get an error on line 587:
```
await expect(errorMessage).toBeVisible();
```

I think the issue is that the first `// Mock a slow response` is not a streamed response.
If you use the same mocked streamed response as `// Mock completion start` on line 591 it successfully asserts the warning.